test/cpp/naming/resolver_component_test.cc

```diff
@@ -46,41 +47,83 @@ extern "C" {
 }
 
 using std::vector;
+using testing::UnorderedElementsAreArray;
 
 DEFINE_string(target_name, "", "");
-DEFINE_bool(expect_target_is_balancer, true, "");
 DEFINE_string(expected_addrs, "", "");
 DEFINE_string(expected_chosen_service_config, "", "");
 DEFINE_string(local_dns_server_address, "", "");
+DEFINE_string(expected_lb_policy, "", "");
 
-namespace grpc {
-namespace testing {
+namespace {
 
-static vector<std::string> parse_expected_addrs(const char *expected_addrs) {
-  std::vector<std::string> out;
+class GrpcLBAddress final {
+ public:
+  GrpcLBAddress(std::string address, bool is_balancer)
+      : is_balancer(is_balancer), address(address) {}
 
-  char *p = (char *)expected_addrs;
+  bool operator<(const GrpcLBAddress &other) const {
+    int comparison = this->address.compare(other.address);
+    if (comparison != 0) {
+      return comparison < 0 ? true : false;
+    }
+    return this->is_balancer == false && other.is_balancer == true;
+  }
+
+  bool is_balancer;
+  std::string address;
+};
+
+bool convert_string_to_bool(std::string bool_str) {
+  if (gpr_stricmp(bool_str.c_str(), "true") == 0) {
+    return true;
+  } else if (gpr_stricmp(bool_str.c_str(), "false") == 0) {
+    return false;
+  } else {
+    gpr_log(GPR_ERROR,
+            "invalid expected_addrs_list entry: expected false or true, got %s",
+            bool_str.c_str());
+    abort();
+  }
+}
+
+vector<GrpcLBAddress> parse_expected_addrs(std::string expected_addrs) {
+  std::vector<GrpcLBAddress> out;
 
-  if (strlen(p) != 0) {
-    for (;;) {
-      char *comma = strchr(p, ',');
-      if (comma) {
-        out.emplace_back(p, comma);
-        p = comma + 1;
-      } else {
-        out.emplace_back(p);
-        break;
-      }
+  while (expected_addrs.size() != 0) {
+    size_t next_comma = expected_addrs.find(",");
+    std::string next_addr;
```

Comment:
Please declare variables right before they are used, rather than declaring them all at once at the top of a function.